### ISSUE TITLE
Wwise SDK setup blocks Pre-built SDK O3DE build

### ISSUE THREAD

**Issue by LB-ArturZieba** [o3de/o3de#13229]
**Describe the bug**
If a Wwise SDK is present on the machine and engine configuration picks it up, the Pre-built SDK build will fail. It should be noted that on the Source configuration the build goes through with the same Wwise SDK present.

Part of the Command Line log of building with failures (for full log please refer to the **Command Line log** section):
```
  DccScriptingInterface.Editor.vcxproj -> C:\o3de\build\windows_vs2019\bin\profile\DccScriptingInterface.Editor.dll
C:\o3de\Gems\AudioEngineWwise\Code\Source\AudioEngineWwiseGemSystemComponent.cpp(124,40): error C2039: 'Descriptor': is not a member of 'Audio::AudioImplAllocator' [C:\o3de\build\windows_vs2019\External\AudioEngineWwise-b
ec3ca47\Code\AudioEngineWwise.Static.vcxproj]
              Audio::AudioImplAllocator::Descriptor allocDesc;
                                         ^
  C:\o3de\Gems\AudioSystem\Code\Include\Engine\AudioAllocators.h(29): note: see declaration of 'Audio::AudioImplAllocator'
      class AudioImplAllocator final
C:\o3de\Gems\AudioEngineWwise\Code\Source\AudioEngineWwiseGemSystemComponent.cpp(124,51): error C2065: 'Descriptor': undeclared identifier [C:\o3de\build\windows_vs2019\External\AudioEngineWwise-bec3ca47\Code\AudioEngineW
wise.Static.vcxproj]
              Audio::AudioImplAllocator::Descriptor allocDesc;
                                                    ^
C:\o3de\Gems\AudioEngineWwise\Code\Source\AudioEngineWwiseGemSystemComponent.cpp(124,51): error C2146: syntax error: missing ';' before identifier 'allocDesc' [C:\o3de\build\windows_vs2019\External\AudioEngineWwise-bec3ca
47\Code\AudioEngineWwise.Static.vcxproj]
              Audio::AudioImplAllocator::Descriptor allocDesc;
                                                    ^
C:\o3de\Gems\AudioEngineWwise\Code\Source\AudioEngineWwiseGemSystemComponent.cpp(124,51): error C2065: 'allocDesc': undeclared identifier [C:\o3de\build\windows_vs2019\External\AudioEngineWwise-bec3ca47\Code\AudioEngineWw
ise.Static.vcxproj]
              Audio::AudioImplAllocator::Descriptor allocDesc;
                                                    ^
C:\o3de\Gems\AudioEngineWwise\Code\Source\AudioEngineWwiseGemSystemComponent.cpp(127,13): error C2065: 'allocDesc': undeclared identifier [C:\o3de\build\windows_vs2019\External\AudioEngineWwise-bec3ca47\Code\AudioEngineWw
ise.Static.vcxproj]
              allocDesc.m_allocationRecords = true;
              ^
C:\o3de\Gems\AudioEngineWwise\Code\Source\AudioEngineWwiseGemSystemComponent.cpp(128,13): error C2065: 'allocDesc': undeclared identifier [C:\o3de\build\windows_vs2019\External\AudioEngineWwise-bec3ca47\Code\AudioEngineWw
ise.Static.vcxproj]
              allocDesc.m_heap.m_numFixedMemoryBlocks = 1;
              ^
C:\o3de\Gems\AudioEngineWwise\Code\Source\AudioEngineWwiseGemSystemComponent.cpp(129,13): error C2065: 'allocDesc': undeclared identifier [C:\o3de\build\windows_vs2019\External\AudioEngineWwise-bec3ca47\Code\AudioEngineWw
ise.Static.vcxproj]
              allocDesc.m_heap.m_fixedMemoryBlocksByteSize[0] = poolSize;
              ^
C:\o3de\Gems\AudioEngineWwise\Code\Source\AudioEngineWwiseGemSystemComponent.cpp(131,13): error C2065: 'allocDesc': undeclared identifier [C:\o3de\build\windows_vs2019\External\AudioEngineWwise-bec3ca47\Code\AudioEngineWw
ise.Static.vcxproj]
              allocDesc.m_heap.m_fixedMemoryBlocks[0] = AZ::AllocatorInstance<AZ::OSAllocator>::Get().Allocate(
              ^
C:\o3de\Gems\AudioEngineWwise\Code\Source\AudioEngineWwiseGemSystemComponent.cpp(132,17): error C2065: 'allocDesc': undeclared identifier [C:\o3de\build\windows_vs2019\External\AudioEngineWwise-bec3ca47\Code\AudioEngineWw
ise.Static.vcxproj]
                  allocDesc.m_heap.m_fixedMemoryBlocksByteSize[0], allocDesc.m_heap.m_memoryBlockAlignment);
                  ^
C:\o3de\Gems\AudioEngineWwise\Code\Source\AudioEngineWwiseGemSystemComponent.cpp(132,66): error C2065: 'allocDesc': undeclared identifier [C:\o3de\build\windows_vs2019\External\AudioEngineWwise-bec3ca47\Code\AudioEngineWw
ise.Static.vcxproj]
                  allocDesc.m_heap.m_fixedMemoryBlocksByteSize[0], allocDesc.m_heap.m_memoryBlockAlignment);
                                                                   ^
  RecastNavigation.Editor.Tests.vcxproj -> C:\o3de\build\windows_vs2019\bin\profile\RecastNavigation.Editor.Tests.dll
C:\o3de\Gems\AudioEngineWwise\Code\Source\AudioEngineWwiseGemSystemComponent.cpp(134,70): error C2065: 'allocDesc': undeclared identifier [C:\o3de\build\windows_vs2019\External\AudioEngineWwise-bec3ca47\Code\AudioEngineWw
ise.Static.vcxproj]
              AZ::AllocatorInstance<Audio::AudioImplAllocator>::Create(allocDesc);
                                                                       ^
```

**Prerequisites**
* Wwise SDK installed (verified with version 2021.1.7601).

**Steps to reproduce**
1. Follow Pre-built SDK configuration build instructions from the https://www.o3de.org/docs/welcome-guide/setup/setup-from-github/building-windows/ documentation page.

**Expected behavior**
Engine is built.

**Actual behavior**
Build fails, o3de-install folder is not created.

**Command Line log**
[WwiseLog.txt](https://github.com/o3de/o3de/files/10021236/WwiseLog.txt)

**Found in Branch**
Development (5234076)

**Desktop**
 - Device: PC
 - OS: Windows
 - Version 11
 - CPU AMD Ryzen 5 3600
 - GPU NVIDIA GeForce RTX 2060 SUPER
 - Memory 16GB

**Comment by amzn-phist** [comment]
This is due to a recent large change to allocators that was merged yesterday.  I'll take a look.

**Comment by LB-ArturZieba** [comment]
Issue verified as fixed on the Development (daed680) build.
